gnoi/os/client.go

```diff
@@ -28,11 +32,129 @@ type Client struct {
 	client pb.OSClient
 }
 
+const chunkSize = 5000000
+
 // NewClient returns a new OS service client.
 func NewClient(c *grpc.ClientConn) *Client {
 	return &Client{client: pb.NewOSClient(c)}
 }
 
+// Install invokes the Install RPC for the OS service.
+func (c *Client) Install(ctx context.Context, imgPath, version string, printStatus bool, validateTimeout time.Duration) error {
```

Comment:
i think this looks nice, however don't assume future readers will be all familiar with the logic you have implemented, specifically because this function is very long, add comments explaining each section briefly but in a descriptive way. A good example would be:

// Check versioning using RPC X

// Transferring content asynchronously using goroutine

// Checking buffer while streaming content

// Conferming transfer

etc...